contracts/protocol/libraries/logic/PositionMoverLogic.sol

```diff
@@ -148,4 +157,127 @@ library PositionMoverLogic {
             })
         );
     }
+
+    function executeClaimOtherdeedAndSupply(
+        mapping(address => DataTypes.ReserveData) storage reservesData,
+        mapping(uint256 => address) storage reservesList,
+        DataTypes.UserConfigurationMap storage userConfig,
+        DataTypes.OtherdeedClaimParams memory params
+    ) external returns (uint256) {
+        DataTypes.ReserveData storage reserve = reservesData[params.otherdeed];
+        DataTypes.ReserveCache memory reserveCache = reserve.cache();
+
+        ValidationLogic.validateWithdrawERC721(
+            reservesData,
+            reserveCache,
+            params.otherdeed,
+            params.otherdeedIds
+        );
+
+        DataTypes.TimeLockParams memory timeLockParams;
+
+        (
+            uint64 oldCollateralizedBalance,
+            uint64 newCollateralizedBalance
+        ) = INToken(reserveCache.xTokenAddress).burn(
+                msg.sender,
+                address(this),
+                params.otherdeedIds,
+                timeLockParams
+            );
+
+        bool isWithdrawCollateral = (newCollateralizedBalance <
+            oldCollateralizedBalance);
+
+        if (isWithdrawCollateral) {
+            if (newCollateralizedBalance == 0) {
+                userConfig.setUsingAsCollateral(reserve.id, false);
+                emit ReserveUsedAsCollateralDisabled(
+                    params.otherdeed,
+                    msg.sender
+                );
+            }
+        }
+
+        IERC721(params.otherdeed).setApprovalForAll(params.vessel, true);
+
+        if (params.kodaIds.length == 0) {
+            IVesselClaim(params.vessel).claimVessels(params.otherdeedIds);
+        } else {
+            IVesselClaim(params.vessel).claimVesselsAndKodas(
+                params.otherdeedIds,
+                params.kodaIds,
+                params.kodaOtherdeedIds,
+                params.merkleProofs
+            );
+        }
+
+        IERC721(params.otherdeed).setApprovalForAll(params.vessel, false);
+
+        SupplyLogic.executeSupplyERC721(
+            reservesData,
+            userConfig,
+            DataTypes.ExecuteSupplyERC721Params({
+                asset: params.otherdeedExpanded,
+                tokenData: _getTokenData(params.otherdeedIds),
```

Comment:
if there is any otherdeedExpanded... stuck in our pool. The others can get them

Reply:
how would that happen? the `claimVessels` would fail if the caller doesn't own the the otherdeeds.

Reply:
never mind, I was thinking using wrong vessel address, but since we need to supply. I think it's ok